uitools/import/Esri/ArcGISRuntime/Toolkit/FloorFilter.qml

```diff
@@ -0,0 +1,599 @@
+/*******************************************************************************
+ *  Copyright 2012-2022 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+import Esri.ArcGISRuntime.Toolkit.Controller 100.14
+import QtQuick 2.12
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+import QtQml.Models 2.15
+import QtGraphicalEffects 1.12
+
+/*!
+  \qmltype FloorFilter
+  \inqmlmodule Esri.ArcGISRuntime.Toolkit
+  \ingroup ArcGISQtToolkitUiQmlViews
+  \since 100.14
+  \brief Allows to display and filter the available floor aware layers in the current \c GeoMap.
+
+  The FloorFilter allows the interaction with the available floor aware layers. A user can select from a list of sites which presents
+  their facilities. Once a facility is chosen, it is possible to toggle between its levels which will show them on the \c GeoView.
+  2D maps and 3D scenes are supported.
+  Example code in the QML API (C++ API might differ):
+  \snippet qml_quick/src/demos/FloorFilterDemoForm.qml Set up Floor Filter QML
+*/
+Control {
+    id: floorFilter
+
+    /*!
+      \qmlproperty GeoView geoView
+      \brief The \c GeoView for this tool. Should be a \c SceneView or \c MapView.
+    */
+    property var geoView
+
+    /*!
+      \qmlproperty FloorFilterController controller
+      \brief The controller handles binding logic between the FloorFilter, \c GeoModel, \c FloorManager and
+      the flooraware layers.
+      Default is \l{FloorFilterController}
+    */
+    property var controller: FloorFilterController {}
+
+    /*!
+      \qmlproperty enumeration updateLevelsMode
+      \brief The mode to use for updating levels visibility.
+      Default is \c {FloorFilterController.UpdateLevelsMode.AllLevelsMatchingVerticalOrder}.
+      \sa {FloorFilterController} {FloorFilterController.updateLevelsMode}
+    */
+    property int updateLevelsMode: controller.updateLevelsMode
+
+    property bool hideSiteFacilityButton: false
```

Comment:
I will keep in mind the rule of thumb. 
Regarding this property, I had a look at the design and could not find anything about it. I am planning to remove it because I don't see the usage: if the property is set to false, it will hide the `buildingMenuButton` which is the one that opens up the listView.